apps/core/scenes/resultDetail/ResultDetailItinerary.js

```diff
@@ -0,0 +1,106 @@
+// @flow
+
+import * as React from 'react';
+import { Text, Loader } from '@kiwicom/universal-components';
+import {
+  graphql,
+  createRefetchContainer,
+  type RelayRefetchProp,
+} from '@kiwicom/margarita-relay';
+
+import type { ResultDetailItinerary_data as ResultDetailItineraryType } from './__generated__/ResultDetailItinerary_data.graphql';
+
+type Props = {|
+  +data: ?ResultDetailItineraryType,
+  +relay: RelayRefetchProp,
+|};
+
+const REFETCH_INTERVAL = 5 * 1000;
+const REFETCH_TIMEOUT = 3 * 60 * 1000;
+
+class ResultDetailItinerary extends React.Component<Props> {
+  componentDidMount() {
+    this.tryStartItineraryCheck();
+  }
+
+  componentDidUpdate(prevProps) {
+    if (
+      this.props.data?.checkItinerary?.itineraryChecked !==
+      prevProps.data?.checkItinerary?.itineraryChecked
+    ) {
+      this.stopItineraryCheck();
+      this.tryStartItineraryCheck();
+    }
+  }
+
+  componentWillUnmount() {
+    this.stopItineraryCheck();
+  }
+
+  checkInterval: IntervalID;
+  checkTimeout: TimeoutID;
+
+  tryStartItineraryCheck = () => {
+    if (
+      this.props.data?.checkItinerary?.itineraryValid &&
+      !this.props.data?.checkItinerary?.itineraryChecked
+    ) {
+      this.checkInterval = setInterval(() => {
+        this.props.relay.refetch();
+      }, REFETCH_INTERVAL);
+
+      this.checkTimeout = setTimeout(() => {
+        this.itineraryCheckTimeout();
+      }, REFETCH_TIMEOUT);
+    }
+  };
+
+  stopItineraryCheck = () => {
+    clearInterval(this.checkInterval);
+    clearTimeout(this.checkTimeout);
+  };
+
+  itineraryCheckTimeout = () => {
+    // @TODO - show timeout error message/content
+    this.stopItineraryCheck();
+  };
+
+  render() {
+    if (!this.props.data?.checkItinerary?.itineraryValid) {
+      return <Text>Not valid booking</Text>;
+    }
+
+    return (
+      <>
+        {!this.props.data?.checkItinerary?.itineraryChecked && (
+          <Loader size="small" />
+        )}
+        <Text>
+          {`Checked: ${
+            this.props.data?.checkItinerary?.itineraryChecked ? 'YES' : 'NO'
+          }`}
+        </Text>
+      </>
+    );
+  }
+}
+
+export default createRefetchContainer(
+  ResultDetailItinerary,
+  {
+    data: graphql`
+      fragment ResultDetailItinerary_data on RootQuery
```

Comment:
Maybe we could name it more specifically, not just `data`

Reply:
I've been thinking about it. And `data` seems to me most universal. Because it will contain mixed content like validity and itinerary data.
But I'm definitely open to any other suggestions.

Reply:
OK, let's keep it. I also couldn't find better name